core/src/test/java/io/parsingdata/metal/data/SourceAndSliceTest.java

```diff
@@ -89,26 +88,24 @@ private boolean compareDataSlices(byte[] data, int offset) {
 
     @Test
     public void readBeyondEndOfSource() throws IOException {
-        thrown.expect(IOException.class);
+        thrown.expect(RuntimeException.class);
```

Comment:
Same here and down below, this tests for a whole slew of exceptions which hopefully will never be thrown.